### PR TITLE
periodic billing - update Transaction Billing Data

### PR DESCRIPTION
**Pull request by ClaudeROSSI**
The periodic billing operation now updates the billing data of each transaction in order to be able to show the final invoice number when looking at the transaction history.

### CODE REVIEW

File: src/integration/billing/stripe/StripeBillingIntegration.ts [hunk 1]
```diff
@@ -20,6 +20,7 @@ import { ServerAction } from '../../../types/Server';
 import Stripe from 'stripe';
 import StripeHelpers from './StripeHelpers';
 import Transaction from '../../../types/Transaction';
+import TransactionStorage from '../../../storage/mongodb/TransactionStorage';
```
Comment (ClaudeROSSI) [inline src/integration/billing/stripe/StripeBillingIntegration.ts:23]: @LucasBrazi06 : This introcuce a direct dependency between the strip concrete implementation and the TransactionStorage. Is this ok with you?

File: src/integration/billing/stripe/StripeBillingIntegration.ts [hunk 2]
```diff
@@ -410,9 +411,36 @@ export default class StripeBillingIntegration extends BillingIntegration {
     await StripeHelpers.updateInvoiceAdditionalData(this.tenantID, billingInvoice, operationResult);
     // Send a notification to the user
     void this.sendInvoiceNotification(billingInvoice);
+    if (FeatureToggles.isFeatureActive(Feature.BILLING_ASYNC_UPDATE_TRANSACTION)) {
+      await this._updateTransactionsBillingData(billingInvoice);
+    }
     return billingInvoice;
   }
 
+  private async _updateTransactionsBillingData(billingInvoice: BillingInvoice): Promise<void> {
```
Comment (ClaudeROSSI) [inline src/integration/billing/stripe/StripeBillingIntegration.ts:420]: @LucasBrazi06 : I'm starting some private methods with an "_" to make it clear that this is a somehow a core feature and that the method is supposed to be called  in a specific context (e.g.: a scheduled job or an async task). 
Is this ok?

File: src/integration/billing/stripe/StripeBillingIntegration.ts [hunk 3]
```diff
@@ -410,9 +411,36 @@ export default class StripeBillingIntegration extends BillingIntegration {
...
+      try {
+        const transaction = await TransactionStorage.getTransaction(this.tenantID, Number(transactionID));
+        // Update Billing Data
+        transaction.billingData.stop.invoiceStatus = billingInvoice.status;
+        transaction.billingData.stop.invoiceNumber = billingInvoice.number;
+        transaction.billingData.lastUpdate = new Date();
+        // Save
+        await TransactionStorage.saveTransaction(this.tenantID, transaction);
+      } catch (error) {
+        // catch stripe errors and send the information back to the client
```
Comment (jerome-benoit) [inline src/integration/billing/stripe/StripeBillingIntegration.ts:432]:
```suggestion
        // Catch stripe errors and send the information back to the client
```

**Review by jerome-benoit — APPROVED** [review]

File: src/integration/billing/stripe/StripeBillingIntegration.ts (hunk 1 above)
Comment (LucasBrazi06) [reply src/integration/billing/stripe/StripeBillingIntegration.ts:23]: I think we already discussed it, right?
Ohterwise, Stripe -> Storage dependency does not hurt me.

File: src/integration/billing/stripe/StripeBillingIntegration.ts (hunk 2 above)
Comment (LucasBrazi06) [reply src/integration/billing/stripe/StripeBillingIntegration.ts:421]: Usually we avoid the '_' in front of the private method.
Maybe we can talk about it?

**Pull request merged**